src/positions/types.ts

```diff
@@ -60,11 +60,20 @@ export interface Shortcut {
   category?: 'claim'
 }
 
+export type ShortcutStatus =
+  | 'idle'
+  | 'loading'
+  | 'pendingAccept'
+  | 'accepting'
+  | 'acceptingInBackground'
```

Comment:
yes but just in the background, maybe there is a better way of doing it - if an approval takes a long time, we need to keep displaying the reward as loading on the claim button (on the screen itself) so we can keep it at the "accepting" state but we can't mark it as "idle" for example. if the approval will take several seconds and the user wants to initiate claiming another reward, they will end up with another reward in the "pendingAccept" state and it won't be clear which tx details we should display in the bottom sheet if we don't transition the first reward into some other state...

Reply:
Can't we use `accepting` for both?

Reply:
yes - maybe i overthought the problem. but if an approval can take 5 seconds, i assumed that in that time a user can dismiss the sheet and start claiming another reward. i used this "acceptingInBackground" to ensure that there can only be one tx in the "pendingAccept" or "accept" states, but i guess we could also achieve this by tracking the id of the reward

Reply:
oh i should add, some of the struggles comes from wanting to persist the bottom sheet during the "accepting" phase. now i'm questioning if that is actually a requirement :| should we just dismiss the bottom sheet after hitting the "allow" button, or persist the bottom sheet with the loading state until manually dismissed / the tx finishes?

Reply:
Good question, seems we could do both, if we dismiss right after tapping the allow button, it feels the UX would be better, but it's more likely the user will try too claim other rewards while the first claim is still happening.
Not sure if this could cause issues with nonces though.

Reply:
yeah, i recall that we intended the user to only have one in flight reward at a time. since the bottom sheet is dismissable though, it's not impossible for them to trigger another but having the bottom sheet persist might reduce the likelihood. having the persisting bottom sheet is also the pattern we use for wallet connect which is why my brain defaulted to that